### PR TITLE
Test/64/test slider input

### LINKED ISSUE #64: (SP: 1) Write tests for slider with input component
Writing tests don't forget to mock elements if neccessary.

- [ ] it should renders correctly
- [ ] it should call onChange when slider is moved
- [ ] it should update inputValue correctly when input value is empty
- [ ] it should update prices when input is blurred and input is greater than max value
- [ ] it should not update prices when input is blurred and value in input has not changed

### PR DESCRIPTION
**Pull request by mistercapxx**
Slider Input tests done
Screenshot 2024-07-17 at 13 54 28

### CODE REVIEW

**Review by Yaroslav-Yakovlev — APPROVED** [review]

File: src/constants/translations/en/guest-home-page.json
Comment (miller-113) [inline src/constants/translations/en/guest-home-page.json:1]: why did you change constants/translations/en/guest-home-page.json file?

**Comment by mistercapxx** [comment]
yes i just copypasted the same json staff from the previous merged project.
Just to resolve conflicts that happened when i wanted to PR my task.

Have a look at json itself :

{
"welcomeBlock": {
"description": "An educational platform that offers professional courses,
empowering students with essential skills for career success.",
"getStarted": "Get started for free"
},
"accordion": {
"flexibleLocation": {
"title": "Flexible Location",
"description": "Space2Study platform has no boundaries and can be used
anywhere in the world, all you need is aspiration."
},
"individualTime": {
"title": "Individual Time",
"description": "Space2Study platform provides the opportunity to study
lessons at any time convenient for you, pause and continue again."
},
"freeChoiceOfTeachers": {
"title": "Free Choice of Tutors",
"description": "Space2Study platform has a wide range of criteria and will
help you find the best tutor for education."
},
"digitalCommunication": {
"title": "Digital Communication",
"description": "Space2Study platform offers extensive opportunities for
meaningful communication not only with tutors but also with students."
}
},
"whatCanYouDo": {
"title": "What can you do",
"description": "Whether you’re looking for a tutor or to looking become one
– you’ve come to the right place.",
"learn": {
"title": "Learn from experts",
"description": "Learning from experts is a shortcut to mastery. Their
knowledge and experience can guide you, unlocking your potential. Don't put
off until tomorrow. Embrace it! 🌟",
"actionLabel": "Become a student"
},
"teach": {
"title": "Share your experience",
"description": "Sharing your experience as an expert will be incredibly
rewarding, helping students unlock their pоtential and achieve their goals.
Don't waste your time 📚",
"actionLabel": "Become a tutor"
}
},
"howItWorks": {
"title": "How it works",
"learnFromExperts": "Learn from Experts",
"shareYourExperience": "Share your Experience",
"student": {
"signUp": {
"title": "Sign Up",
"description": "Registering on the Space2Study platform is very simple,
just enter your email or use your Google account."
},
"selectATutor": {
"title": "Select a Tutor",
"description": "Finding a tutor is very easy, select a subject from the
list, and then pick the tutor, or enter his name and find directly."
},
"sendRequest": {
"title": "Send Request",
"description": "Write a request to the tutor in two clicks, where you
indicate/confirm the price and the desired level of training."
},
"startLearning": {
"title": "Start Learning",
"description": "After confirming the request, all training opportunities
for the chosen course will open for you."
}
},
"tutor": {
"signUp": {
"title": "Sign Up",
"description": "Registering on the Space2Study platform is very simple,
just enter your email or use your Google account."
},
"createATutorAccount": {
"title": "Create a Tutor Account",
"description": "Creating a tutor account is easy, you just need to select a
subject, upload teaching materials and create an offer."
},
"getNewStudents": {
"title": "Get New Students",
"description": "Create fruitful cooperation with your students, confirming
their requests, and start teaching."
},
"receiveFeedbacks": {
"title": "Receive Feedbacks",
"description": "Receive positive feedback for your work to become popular
and get even more satisfied students."
}
}
},
"whoWeAre": {
"title": "Who we are",
"description": "SpaceToStudy platform is a place where experts share their
knowledge and students can learn from the best. Here you can find the
proper training course, find a tutor, or find a student. SpaceToStude
doesn’t receive any commission for classes between a tutor and a student,
that’s why education with us becomes even closer and more financially
available."
},
"faq": {
"title": "Frequently Asked Questions",
"description":"Everything you need to know about learning journey in
Space2Study as a student",
"howToFindTutor" :
{
"title": "How to find a tutor",
"description": "It is a long established fact that a reader will be
distracted by the readable content of a page when looking at its layout.
The point of using Lorem Ipsum is that it has a more-or-less normal
distribution of letters, as opposed to using 'Content here, content here',
making it look like readable English."
},
"howToBookLesson":
{
"title": "How to book a lesson",
"description": "Booking a lesson is easy. Simply choose your preferred
tutor and schedule a time that works for you."
},
"rulesForStudents":
{
"title":"Rules for students",
"description":"Students are expected to attend lessons on time and complete
any assignments given by their tutors."
},
"howToPayForLessons":
{
"title":"How you can pay for lessons",
"description":"Payments can be made through our secure online payment
system using a credit card or PayPal."
}
}
}

**Review by kryzanivska-nastya — APPROVED** [review]

File: src/constants/translations/en/guest-home-page.json [hunk 1]
```diff
@@ -19,7 +56,7 @@
   },
   "whatCanYouDo": {
     "title": "What can you do",
-    "description": "Whether you’re looking for a tutor or looking to become one – you’ve come to the right place.",
+    "description": "Whether you’re looking for a tutor or to looking become one – you’ve come to the right place.",
```
Comment (CallMeMaverick) [inline src/constants/translations/en/guest-home-page.json:59]: I think you shouldn't have changed the order. "looking to" was certainly a correct one

**Review by Yaroslav-Yakovlev — COMMENTED** [review]
There should be no commented lines in the code

File: src/tests/unit/components/slider-with-input/SliderWithInput.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        defaultValue={defaults.defaultValue}
+        max={defaults.max}
+        min={defaults.min}
+        onChange={mockOnChange}
+        title={defaults.title}
+      />
+    )
+
+    const sliderElement = screen.getByRole('slider')
+    // console.log(sliderElement);
```
Comment (nrevutska) [inline src/tests/unit/components/slider-with-input/SliderWithInput.spec.jsx:33]: this comment shouldn't be here

File: src/tests/unit/components/slider-with-input/SliderWithInput.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    fireEvent.change(sliderElement, { target: { value: 75 } })
+    // console.log('mock:',mockOnChange.mock.calls);
```
Comment (nrevutska) [inline src/tests/unit/components/slider-with-input/SliderWithInput.spec.jsx:35]: this comment shouldn't be here

File: src/tests/unit/components/slider-with-input/SliderWithInput.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        defaultValue={defaults.defaultValue}
+        max={defaults.max}
+        min={defaults.min}
+        onChange={mockOnChange}
+        title='Range Title'
+      />
+    )
+
+    const inputElement = screen.getByRole('textbox')
+    fireEvent.change(inputElement, { target: { value: '150' } })
```
Comment (nrevutska) [inline src/tests/unit/components/slider-with-input/SliderWithInput.spec.jsx:71]: better use userEvent in place of fireEvent https://testing-library.com/docs/user-event/intro
image

**Review by nrevutska — CHANGES_REQUESTED** [review]
Consider all the reviews of this pr, please

**Pull request closed without merging**